### ISSUE TITLE
PDF files not syncing properly

### ISSUE THREAD

**Issue by eillarra** [owncloud/core#24312]
### Steps to reproduce
1. Make a clean installation on Scientific Linux 7.2 (RHEL7)
2. Try to sync a bunch of files
3. PDF files are not synced (more background [here](https://github.com/owncloud/client/issues/4542#issuecomment-197774868))

### Expected behaviour
PDF files should sync like the rest of the files.

### Actual behaviour
PDF files are not synced. I have a folder with around 20 PDF files, and only 2 are actually synced (I can see them in the server). But even this 2 files are not accessible via the website (the PDF viewer doesn't work).

On the other hand, I can upload some of the failing files directly via the web interface. But not all. So, let's say I have 40 files (PDFs, Word documents, etc.):

- If I try to upload them via the Client, 15 PDF files fail (only PDF files).
- If I upload them using the browser, only 2 PDF files fail, but only 25 of the files are synced locally (the 15 PDF files that will fail using the Client are not synced either). So I have 38/40 files online, and 25/40 in my local machine.

I thought this was a problem caused by Scientific Linux 6, but I reinstalled ownCloud to a brand new machine today, and the problem persists.

### Server configuration
**Operating system**: Scientific Linux 7.2 (RHEL7)
**Web server:** Apache/2.4.6 (Scientific Linux)
**Database:** MySQL 5.5.47-MariaDB
**PHP version:** PHP 5.4.16
**ownCloud version:** ownCloud 9.0.1 (stable)

**Updated from an older ownCloud or fresh install:** fresh install

**Where did you install ownCloud from:** yum

**Signing status (ownCloud 9.0 and above):** 
```
No errors have been found.
```

**List of activated apps:**
```
Enabled:
  - activity: 2.2.1
  - dav: 0.1.6
  - federatedfilesharing: 0.1.0
  - files: 1.4.4
  - files_pdfviewer: 0.8
  - files_sharing: 0.9.1
  - files_texteditor: 2.1
  - files_trashbin: 0.8.0
  - files_versions: 1.2.0
  - files_videoplayer: 0.9.8
  - firstrunwizard: 1.1
  - notifications: 0.2.3
  - provisioning_api: 0.4.1
  - systemtags: 0.2
  - templateeditor: 0.1
  - updatenotification: 0.1.0
Disabled:
  - comments
  - encryption
  - external
  - federation
  - files_external
  - gallery
  - user_external
  - user_ldap
```

**The content of config/config.php:**

```
{
    "system": {
        "updatechecker": false,
        "instanceid": "ocs25u9pjv0t",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "www.example.net"
        ],
        "datadirectory": "\/mnt\/owncloud\/data",
        "overwrite.cli.url": "https:\/\/www.example.net\/cloud",
        "dbtype": "mysql",
        "version": "9.0.1.3",
        "dbname": "owncloud",
        "dbhost": "127.0.0.1",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "logtimezone": "UTC",
        "installed": true,
        "memcache.local": "\\OC\\Memcache\\APC",
        "loglevel": 0,
        "skeletondirectory": "\/mnt\/owncloud\/skeleton",
        "enabledPreviewProviders": [
            "OC\\Preview\\PNG",
            "OC\\Preview\\JPEG",
            "OC\\Preview\\GIF",
            "OC\\Preview\\BMP",
            "OC\\Preview\\XBitmap",
            "OC\\Preview\\MP3",
            "OC\\Preview\\TXT",
            "OC\\Preview\\MarkDown",
            "OC\\Preview\\PDF"
        ],
        "maintenance": false,
        "theme": ""
    }
}
```

**Are you using external storage, if yes which one:** cifs mount

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** no

### Client configuration
**Browser:** Chrome 48.0.2564.103 (64-bit)
**Operating system:** OS X El Capitan 10.11.4 (15E65)

#### Web server error log
```
None
```

#### ownCloud log (data/owncloud.log)
```
None
```

**Comment by nickvergessen** [comment]
Can you provide a package with failing pdf files?

**Comment by PVince81** [comment]
Something to note: encryption is enabled.
Find out whether the failing PDFs are bigger than 10 MB, size after which chunking is used.

**Comment by eillarra** [comment]
@nickvergessen Here you can find a selection of files: https://www.dropbox.com/sh/zieac373py5awv3/AACO-XiwerINzS_Izzn5tDo_a?dl=0

- Files in the `failing-examples` folder always fail syncing when you use the ownCloud Client (OS X). But, 3 of the 4 files can be uploaded using the web browser: only the Forbes file cannot be uploaded neither using the native Client or the browser (`IT Revolution...`). So, after trying to upload the 4 files from the web browser, I have 3 on the server, and none in my Mac. The web client reports "Missing PDF file." for all the 3 files that show up in the browser. @PVince81 notice that all files are < 10 MB.

- File in the `working-examples` folder can be uploaded directly using both the OS X Client or the web browser.

**Comment by eillarra** [comment]
@PVince81 I have updated the issue, because encryption is not enabled: I have run `occ encryption:status` to check it. When I wrote the issue I was thinking on HTTPS.

**Comment by nickvergessen** [comment]
I just tested and I can upload them all just fine with the sync client and via the web UI.

The files that I uploaded via the web UI also synced down correctly...

So the only thing I can not do is test with OS X or whether dropbox already modified the file/names in any way so that the actual problem got removed.

**Comment by eillarra** [comment]
Here the same files in a ZIP file, created inside the ownCloud folder (the ZIP file does get synced):
[owncloud-core-issue-24312.zip](https://github.com/owncloud/core/files/242019/owncloud-core-issue-24312.zip)

**Comment by nickvergessen** [comment]
Still works, so it might be related to OSX

**Comment by eillarra** [comment]
Some of the users have problems using Windows too. I will try to test this same files in a Windows machine against our ownCloud server. Also, how do you explain that 3 of the 4 files are uploaded correctly to the server using a browser? This is something that happens in Chrome (48.0.2564.103), Firefox (44.0.2) and Safari (9.1).

**Comment by eillarra** [comment]
Windows users have the same problems when uploading the PDF files: are there knows incompatibilities with Scientific Linux or are there key packages that should be installed in my Linux distribution? As far as I know, Scientific Linux is based on RHLE, the preferred distribution for ownCloud...

Example of a standard Windows machine having this same problems:
**OS**: Windows 8 Enterprise
**ownCloud client**: Version 2.1.1 (build 5837)

Same files fail (`failing-examples`) and same working example is correctly sync-ed. As a side note, the file in the `working-examples` folder is synced with the server, but it is not downloaded to my machine (OS X).

**Comment by PVince81** [comment]
PDF files should always sync properly. ownCloud server or client do not have any special handling for such files. In the past, some people also reported having trouble syncing image files. Most of the time it was a web server configuration issue, like some web server module trying to be clever with known file types. You might want to check what web server modules are enabled.

**Comment by eillarra** [comment]
I have deployed a DigitalOcean droplet and tested this files from my computer and everything works fine. So it is not a client problem and the problem is definitely in our server's configuration. Any tips on where we should start checking?

```
 core_module (static)
 mpm_prefork_module (static)
 http_module (static)
 so_module (static)
 auth_basic_module (shared)
 auth_digest_module (shared)
 authn_file_module (shared)
 authn_alias_module (shared)
 authn_anon_module (shared)
 authn_dbm_module (shared)
 authn_default_module (shared)
 authz_host_module (shared)
 authz_user_module (shared)
 authz_owner_module (shared)
 authz_groupfile_module (shared)
 authz_dbm_module (shared)
 authz_default_module (shared)
 ldap_module (shared)
 authnz_ldap_module (shared)
 include_module (shared)
 log_config_module (shared)
 logio_module (shared)
 env_module (shared)
 ext_filter_module (shared)
 mime_magic_module (shared)
 expires_module (shared)
 deflate_module (shared)
 headers_module (shared)
 usertrack_module (shared)
 setenvif_module (shared)
 mime_module (shared)
 dav_module (shared)
 status_module (shared)
 autoindex_module (shared)
 info_module (shared)
 dav_fs_module (shared)
 vhost_alias_module (shared)
 negotiation_module (shared)
 dir_module (shared)
 actions_module (shared)
 speling_module (shared)
 userdir_module (shared)
 alias_module (shared)
 substitute_module (shared)
 rewrite_module (shared)
 proxy_module (shared)
 proxy_balancer_module (shared)
 proxy_ftp_module (shared)
 proxy_http_module (shared)
 proxy_ajp_module (shared)
 proxy_connect_module (shared)
 cache_module (shared)
 suexec_module (shared)
 disk_cache_module (shared)
 cgi_module (shared)
 version_module (shared)
 perl_module (shared)
 php5_module (shared)
 ssl_module (shared)
 dav_svn_module (shared)
 authz_svn_module (shared)
 wsgi_module (shared)
 xsendfile_module (shared)
```

**Comment by PVince81** [comment]
Please note that this is a bug tracker, not a support forum.
For support, check http://owncloud.org/support

Unfortunately I don't know enough about PHP modules to be able to help here.

**Comment by eillarra** [comment]
I was able to debug this a little bit more, and this is what I have found:

- A fresh installation in Ubuntu gives the same problems.
- The problem only occurs when connecting via HTTPS.

So, I have now a ownCloud client connected to ownCloud 9.0.2 on Ubuntu. For testing purposes, I have now two folders, both connected to the same server: one using SSL and another **without SSL**.

Any file I put in the non-SSL folder will be correctly synced with the server, and then downloaded without problems to the SSL folder on my Mac. On the other hand, if I try to add the test files ([owncloud-core-issue-24312.zip](https://github.com/owncloud/core/files/242019/owncloud-core-issue-24312.zip)) **directly in the SSL folder then some files won't be synced with the server**.
